### PR TITLE
Add builtin input() function.

### PR DESCRIPTION
**Pull request by dpgeorge**
This patch adds the builtin input() function.  It's pretty minimal because it uses the existing readline function to capture the input line.

It was requested in #349, and although using UART was found to be a solution to a missing input() function, I think input is useful enough (and has many applications in the class room) to add it.

### CODE REVIEW

**Comment by ZanderBrown** [comment]
Without it the REPL can be kinda limited

File: source/microbit/input.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2013-2016 Damien P. George
```
Comment (markshannon) [inline source/microbit/input.c]: 2013? This took a while to write :smiling_imp:

**Comment by markshannon** [comment]
LGTM and tested. 
@dpgeorge do you want to rebase before I merge?

**Comment by markshannon** [comment]
I forget that github can do fast-forward merges now. Merging.

File: source/microbit/input.c (hunk 1 above)
Comment (dpgeorge) [inline source/microbit/input.c]: :) it's from upstream uPy and was written long ago.

**Pull request merged**